File: app/src/main/java/me/jy/danggi/model/Memo.java
package me.jy.danggi.model;

import java.io.Serializable;
import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Created by JY on 2018-01-12.
 */
@Data
@AllArgsConstructor(staticName = "of")
@NoArgsConstructor
public class Memo implements Serializable{

    private int id;
    private String content;
    private Date writeDate;


    public Memo ( String content ) {
        this.content = content;
    }

    public Memo ( String content, Date writeDate ) {
        this.content = content;
        this.writeDate = writeDate;
    }

    public Memo ( int id, String content ) {
        this.id = id;
        this.content = content;
    }
}

File: app/src/main/java/me/jy/danggi/model/Memo.kt
package me.jy.danggi.model

import io.realm.RealmObject
import io.realm.RealmResults
import io.realm.annotations.LinkingObjects
import io.realm.annotations.PrimaryKey
import java.io.Serializable
import java.util.*

open class Memo : RealmObject(), Serializable {

    @PrimaryKey
    var id: Int= 0
    var content: String = ""
    var writeDate: Date? = null

    @LinkingObjects("memo")
    val widgets: RealmResults<Widget>?=null

}